crates/turborepo-lib/src/process/command.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    collections::BTreeMap,
+    ffi::{OsStr, OsString},
+    process::Stdio,
+};
+
+use itertools::Itertools;
+use turbopath::AbsoluteSystemPathBuf;
+
+/// A command builder that can be used to build both regular
+/// child processes and ones spawned hooked up to a PTY
+pub struct Command {
+    program: OsString,
+    args: Vec<OsString>,
+    cwd: Option<AbsoluteSystemPathBuf>,
+    env: BTreeMap<OsString, OsString>,
+    open_stdin: bool,
+    env_clear: bool,
+}
+
+impl Command {
+    pub fn new(program: impl AsRef<OsStr>) -> Self {
+        let program = program.as_ref().to_os_string();
+        Self {
+            program,
+            args: Vec::new(),
+            cwd: None,
+            env: BTreeMap::new(),
+            open_stdin: false,
+            env_clear: false,
+        }
+    }
+
+    pub fn args<I, S>(&mut self, args: I) -> &mut Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        self.args = args
+            .into_iter()
+            .map(|arg| arg.as_ref().to_os_string())
+            .collect();
+        self
+    }
+
+    pub fn current_dir(&mut self, dir: AbsoluteSystemPathBuf) -> &mut Self {
+        self.cwd = Some(dir);
+        self
+    }
+
+    pub fn envs<I, K, V>(&mut self, vars: I) -> &mut Self
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: AsRef<OsStr>,
+        V: AsRef<OsStr>,
+    {
+        self.env = vars
+            .into_iter()
+            .map(|(k, v)| (k.as_ref().to_os_string(), v.as_ref().to_os_string()))
+            .collect();
+        self
+    }
+
+    pub fn env<K, V>(&mut self, key: K, val: V) -> &mut Self
+    where
+        K: AsRef<OsStr>,
+        V: AsRef<OsStr>,
+    {
+        self.env
+            .insert(key.as_ref().to_os_string(), val.as_ref().to_os_string());
+        self
+    }
+
+    /// Configure if the child process should spawn with stdin or not
+    pub fn open_stdin(&mut self, open_stdin: bool) -> &mut Self {
+        self.open_stdin = open_stdin;
+        self
+    }
+
+    /// Clears the environment variables for the child process
+    pub fn env_clear(&mut self) -> &mut Self {
```

Comment:
Along with above, I think this naming is confusing. It doesn't clear `self.env`, so we might want to think about the naming.

Along similar lines, `open_stdin` takes a `bool` and sets a `bool`. This function takes nothing and sets a `bool`. Should we change one or the other?

Reply:
Similar to above, I'm trying to match the API of stdlib/tokio.

Updated `open_stdin` to no longer take a bool.